pkg/api/orchestratorversiontypes.go

```diff
@@ -1,6 +1,29 @@
 package api
 
-// OrchestratorVersionProfile contains orchestrator version info
+// PoolUpgradeProfile contains pool properties:
+//  - orchestrator type and version
+//  - pool name (for agent pool)
+//  - OS type of the VMs in the pool
+//  - list of applicable upgrades
+type PoolUpgradeProfile struct {
```

Comment:
Why do we need internal API model for this? Couldn't think just we construct and return the public API model. Internal API model mainly exists because ACS Engine needs a standardized model to operation on to generate templates and also to save normalized data in the DB.